canonical-web-and-design/maas-ui
# feat(ui): add a form for editing physical interfaces

## Done

- Add the form for editing physical interfaces to the interfaces table.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View the network details for a machine.
- Open the action menu for a physical interface and click edit.
- Fill out the fields.
- Setting the link speed higher than the interface speed should show an error (on blur).
- Save the form and the interface should update.

## Fixes

Fixes: #2148.
